sc-poc/pkg/apis/core/v1alpha1/pubsub_channel.go

```diff
@@ -23,7 +25,7 @@ type PubsubChannelSpec struct {
 	// Channel describes the name of the pubsub channel
 	Channel string `json:"channel"`
 	// Payload describes the payload schema of the channel
-	Payload map[string]*ChannelSchema `json:"payload,omitempty"`
+	Payload *ChannelSchema `json:"payload,omitempty"`
```

Comment:
This is not creating a proper crd i guess... Can we making this a map ehich we can then mapstructure decode?